bundle/edu.gemini.seqexec.odb/src/main/scala/edu/gemini/seqexec/odb/SeqExecService.scala

```diff
@@ -83,25 +88,43 @@ object SeqExecService {
       }
     }
 
+  private def extractName(obs: ISPObservation): TrySeq[String] = trySeq {
+    obs.getDataObject match {
+      case o: SPObservation => o.getTitle.right
+      case _                => SeqException(new RuntimeException("Cannot read the obs title")).left
+    }
+  }
+
   private def extractSequence(obs: ISPObservation): TrySeq[ConfigSequence] =
     catchingAll {
       ConfigBridge.extractSequence(obs, null, IDENTITY_MAP, true)
     }
 
+  private def extractExecutedDatsets(obs: ISPObservation): TrySeq[Map[Int, ExecutedDataset]] =
+    catchingAll {
+      Option(ObsLog.getIfExists(obs)) match {
+        case Some(obsLog) => obsLog.getAllDatasetRecords.asScala.map(_.exec.dataset).map { d =>
+          d.getIndex -> ExecutedDataset(Instant.ofEpochMilli(d.getTimestamp), d.getDhsFilename)
```

Comment:
It seems that `getIndex` is the step where the file was executed right. There is another `index` on `DatasetLabel`. @swalker2m  Do you know if they are the same?

Reply:
I usually key datasets by their full dataset label, which includes the index.  In reality in the current OCS these must be the same.  That's the source of innumerable problems that motivated the need for a new sequence model.